plugins/inputs/azure_monitor/azure_monitor.go

```diff
@@ -0,0 +1,235 @@
+    metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+    # metrics aggregation type value to collect
+    # can be 'Total', 'Count', 'Average', 'Minimum', 'Maximum'
+    # leave the array empty to collect all aggregation types values for each metric
+    aggregations = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+    
+  # resource target #2 to collect metrics of
+  [[inputs.azure_monitor.resource_target]]
+    resource_id = "<<RESOURCE_ID>>"
+    metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+    aggregations = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+
+  # resource group target #1 to collect metrics of resources under it with resource type
+  [[inputs.azure_monitor.resource_group_target]]
+    # the resource group name
+    resource_group = "<<RESOURCE_GROUP_NAME>>"
+
+    # defines the resources to collect metrics from
+    [[inputs.azure_monitor.resource_group_target.resource]]
+      # the resource type
+      resource_type = "<<RESOURCE_TYPE>>"
+      metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+      aggregations = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+    
+    # defines the resources to collect metrics from
+    [[inputs.azure_monitor.resource_group_target.resource]]
+      resource_type = "<<RESOURCE_TYPE>>"
+      metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+      aggregations = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+      
+  # resource group target #2 to collect metrics of resources under it with resource type
+  [[inputs.azure_monitor.resource_group_target]]
+    resource_group = "<<RESOURCE_GROUP_NAME>>"
+
+    [[inputs.azure_monitor.resource_group_target.resource]]
+      resource_type = "<<RESOURCE_TYPE>>"
+      metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+      aggregations = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+  
+  # subscription target #1 to collect metrics of resources under it with resource type    
+  [[inputs.azure_monitor.subscription_target]]
+    resource_type = "<<RESOURCE_TYPE>>"
+    metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+    aggregations = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+    
+  # subscription target #2 to collect metrics of resources under it with resource type    
+  [[inputs.azure_monitor.subscription_target]]
+    resource_type = "<<RESOURCE_TYPE>>"
+    metrics = [ "<<METRIC>>", "<<METRIC>>" ]
+    aggregations = [ "<<AGGREGATION>>", "<<AGGREGATION>>" ]
+`
+
+func (am *AzureMonitor) Description() string {
+	return "Gather Azure resources metrics using Azure Monitor API"
+}
+
+func (am *AzureMonitor) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init is for setup, and validating config.
+func (am *AzureMonitor) Init() error {
+	if am.azureClients == nil {
+		azureClients, err := receiver.CreateAzureClients(am.SubscriptionID, am.ClientID, am.ClientSecret, am.TenantID)
+		if err != nil {
+			return err
+		}
+
+		am.azureClients = azureClients
+	}
+
+	if err := am.setReceiver(); err != nil {
+		return fmt.Errorf("error setting Azure Monitor receiver: %v", err)
```

Comment:
Can you please use error wrapping here? Same for all other `fmt.Errorf` instances...
```suggestion
		return fmt.Errorf("error setting Azure Monitor receiver: %w", err)
```